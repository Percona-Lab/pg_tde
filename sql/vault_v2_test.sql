CREATE EXTENSION pg_tde;

SELECT pg_tde_add_key_provider_vault_v2('vault-v2','ROOT_TOKEN','http://127.0.0.1:8200','secret',NULL);
SELECT pg_tde_set_master_key('vault-v2-master-key','vault-v2');

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING pg_tde;

INSERT INTO test_enc (k) VALUES (1);
INSERT INTO test_enc (k) VALUES (2);
INSERT INTO test_enc (k) VALUES (3);

SELECT * from test_enc;

DROP TABLE test_enc;

DROP EXTENSION pg_tde;
