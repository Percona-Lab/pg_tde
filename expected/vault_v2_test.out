CREATE EXTENSION pg_tde;
SELECT pg_tde_add_key_provider_vault_v2('vault-v2','ROOT_TOKEN','http://127.0.0.1:8200','secret',NULL);
 pg_tde_add_key_provider_vault_v2 
----------------------------------
                                1
(1 row)

SELECT pg_tde_set_master_key('vault-v2-master-key','vault-v2');
 pg_tde_set_master_key 
-----------------------
 
(1 row)

CREATE TABLE test_enc(
	  id SERIAL,
	  k INTEGER DEFAULT '0' NOT NULL,
	  PRIMARY KEY (id)
	) USING pg_tde;
INSERT INTO test_enc (k) VALUES (1);
INSERT INTO test_enc (k) VALUES (2);
INSERT INTO test_enc (k) VALUES (3);
SELECT * from test_enc;
 id | k 
----+---
  1 | 1
  2 | 2
  3 | 3
(3 rows)

DROP TABLE test_enc;
DROP EXTENSION pg_tde;
